service/migration_manager.cc

```diff
@@ -103,10 +100,6 @@ void migration_manager::init_messaging_service()
         _feature_listeners.push_back(_feat.cluster_supports_cdc().when_enabled(update_schema));
         _feature_listeners.push_back(_feat.cluster_supports_per_table_partitioners().when_enabled(update_schema));
     }
-    _feature_listeners.push_back(_feat.cluster_supports_schema_tables_v3().when_enabled([this] {
-        _cluster_upgraded = true;
-        _wait_cluster_upgraded.broadcast();
-    }));
 
```

Comment:
This patch looks odd since the action that triggers the change isn't dependent on cluster_supports_schema_tables_v3. However, the patch is correct and the code is odd. Here's the v3 change in main.cc:

```c++
            // schema migration, if needed, is also done on shard 0
            db::legacy_schema_migrator::migrate(proxy, db, qp.local()).get();
```

So we couldn't roll back a bad 1.7->2.0 rolling upgrade.